src/Documentation/Getting-Started-With-Orleans/Developing-a-Grain.md

```diff
@@ -38,26 +45,101 @@ public class PlayerGrain : Grain, IPlayerGrain
+```csharp
+public Task GrainMethod2()
+{
+    ...
+    return Task.CompletedTask;
+}
+```
+
+A grain method marked as `async` returns the value directly:
+```csharp
+public async Task<SomeType> GrainMethod3()
+{
+    ...
+    return <variable or constant with result>;
+}
+```
+A "void" grain methods marked as `async` that returns no value simply returns at the end of their execution:
+```csharp
+public async Task GrainMethod4()
+{
+    ...
+    return;
+}
+```
+
+If a grain method receives the return value from another asynchronous method call, to a grain or not, and doesn't need to perform error handling of that call, it can simply return the `Task` it receives from that asynchronous call as its return value:
+```csharp
+public Task<SomeType> GrainMethod5()
+{
+    ...
+    Task<SomeType> task = CallToAnotherGrain();
+    return task;
+}
+```
+Similarly, a "void" grain method can return a `Task` returned to it by another call instead of awaiting it.
+```csharp
+public Task GrainMethod6()
+{
+    ...
+    Task task = CallToAsyncAPI();
+    return task;
+}
+```
+
+### Grain Reference
+
+A Grain Reference is a proxy object that implements the same grain interface as the corresponding grain class.
+It encapsulates a logical identity (type and unique key) of the target grain.
+A grain reference is what is used for making calls to the target grain.
+Each grain reference is for a single grain (a single instance of the grain class), but one can create multiple independent references for the same grain.
+
+Since a grain reference represents a logical identity of the target grain, it is independent from the physical location of the grain, and stays valid even after a complete restart of the system.
+Developers can use grain references like any other .NET object.
+It can be passed to a method, used as a method return value, etc., and even saved to persistent storage. 
 
-A Grain Reference is a proxy object that implements the same grain interface implemented by the corresponding grain class. Using asynchronous messaging, it provides full-duplex communication with other grains, as well as [Orleans Client](/Orleans/Getting-Started-With-Orleans/Clients) code.
-A grain reference can be constructed by passing the identity of a grain to the `GrainFactory.GetGrain<T>()` method, where T is the grain interface. Developers can use grain references like any other .NET object. It can be passed to a method, used as a method return value, etc.
+A grain reference can be obtained by passing the identity of a grain to the `GrainFactory.GetGrain<T>(key)` method, where `T` is the grain interface and `key` is the unique key of the grain within the type. 
 
-The following are examples of how to construct a grain reference of the `IPlayerGrain` interface defined above.
+The following are examples of how to obtain a grain reference of the `IPlayerGrain` interface defined above.
 
 In Orleans Client code:
```

Comment:
How about this version? I don't think we want to show how to construct a client just yet. But I added a link to the next page that will explain what the client is and how to construct it.